src/NHibernate/Logging.cs

```diff
@@ -222,16 +223,16 @@ public void WarnFormat(string format, params object[] args)
 	public class Log4NetLoggerFactory: ILoggerFactory
 	{
 		private static readonly System.Type LogManagerType = System.Type.GetType("log4net.LogManager, log4net");
-		private static readonly Func<string, object> GetLoggerByNameDelegate;
+		private static readonly Func<Assembly, string, object> GetLoggerByNameDelegate;
 		private static readonly Func<System.Type, object> GetLoggerByTypeDelegate;
 		static Log4NetLoggerFactory()
 		{
-			GetLoggerByNameDelegate = GetGetLoggerMethodCall<string>();
+			GetLoggerByNameDelegate = GetGetLoggerByNameMethodCall();
 			GetLoggerByTypeDelegate = GetGetLoggerMethodCall<System.Type>();
 		}
 		public IInternalLogger LoggerFor(string keyName)
 		{
-			return new Log4NetLogger(GetLoggerByNameDelegate(keyName));
+			return new Log4NetLogger(GetLoggerByNameDelegate(typeof(Log4NetLoggerFactory).Assembly, keyName));
```

Comment:
No, because the delegate is running in the context of the `Log4NetLoggerFactory` class, so when `Assembly.GetCallingAssembly()` was called in `log4net`, it only goes one call-level up--back to `NHibernate`.